app/controllers/admin/cards_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class Admin::CardsController < Admin::BaseController
+  before_action :admin_authorize!
+  before_action :set_card, only: %i[show destroy]
+
+  def index
+    @cards = Card.all
+  end
+
+  def new
+    @card = Card.new
+  end
+
+  def create
+    @card = Card.new(card_params)
+    current_user.cards << @card
+    @card.card_assignments.last.expires_at = Time.current
```

Comment:
This has a "race" condition, what happens if this is called twice or another card assignment was created.   Can we get a handle of the new card assignment that you just made.